src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -379,89 +386,217 @@ internal Database(C4Database* c4db)
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
         public IReadOnlyList<Scope> GetScopes()
         {
-            throw new NotImplementedException();
+            GetScopesList();
+            return _scopes?.Values as IReadOnlyList<Scope>;
         }
 
         /// <summary>
-        /// Get a scope object by name. As the scope cannot exist by itself without having a collection, the nil 
-        /// value will be returned if there are no collections under the given scope’s name.
-        /// Note: The default scope is exceptional, and it will always be returned.
+        /// Get a scope object by name. As the scope cannot exist by itself without having a collection, the null 
+        /// value will be returned if there is no collection under the given scope’s name. 
+        /// The default scope is an exception, it will always be returned.
         /// </summary>
-        /// <param name="name"></param>
-        /// <returns></returns>
+        /// <param name="name">The name of the scope</param>
+        /// <returns>scope object with the given scope name</returns>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
         public Scope GetScope(string name)
         {
-            throw new NotImplementedException();
+            Scope scope = null;
+            ThreadSafety.DoLocked(() =>
+            {
+                CheckOpen();
+                if (HasScopeFromLiteCore(name)) {
+                    if (!_scopes.ContainsKey(name)) {
+                        scope = new Scope(this, name);
+                    } else {
+                        scope = _scopes[name];
+                    }
+                }
+            });
+
+            return scope;
         }
 
         /// <summary>
         /// Get all collections of given Scope name.
         /// </summary>
-        /// <returns></returns>
+        /// <param name="scope">The scope of the collections belong to</param>
+        /// <returns>All collections with the given scope name</returns>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
         public IReadOnlyList<Collection> GetCollections(string scope = _defaultScopeName)
         {
-            throw new NotImplementedException();
+            Scope s = null;
+            ThreadSafety.DoLocked(() =>
+            {
+                CheckOpen();
+                s = scope == _defaultScopeName ? GetDefaultScope() : GetScope(scope);
```

Comment:
Small nit: GetScope also has the default scope, so why not just always use GetScope?

Reply:
GetDefaultScope() has much simpler logic, I need to check if LiteCore HasScope in GetScope, but default scope always exists in the database....